### PR TITLE
Willys/qa-integration

### PR DESCRIPTION
**Pull request by wpschaub**
The first or I hope many blueprint discussion posts. This one drills into the QA extension points.

### CODE REVIEW

**Comment by wpschaub** [comment]
@604kev you **must** scan/review this PR and ensure the content is correct - same logic applies to the security automation scans.

**Review by danielrbroderick — DISMISSED** [review]

File: content/azure-pipelines-blueprint-qa-integration.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+# --------------------------------------------------------------------------
+    - task: DotNetCoreCLI@2
+      displayName: 'Run Tests'
+      inputs:
+        command:    'test'
+        projects:   '**/*Tests/*.csproj'
+        arguments:  '--configuration $(buildConfiguration) --no-restore'
+```
+
+It is the responsibility of the software delivery team to define **consistent**, **comprehensive**, and **quality** continuous integration tests to empower us to automate the unite and regression testing while building the artifacts we want to deploy. You can extend the [building-code](https://github.com/WorkSafeBC-Common-Engineering/AzureDevOps.Automation.Pipeline.Templates.v2/blob/master/templates/building-code/building-code.yml) to validate the health of your unit tests by using a tool such as [Stryker](https://stryker-mutator.io/).
```
Comment (MerlinDarkstar) [inline content/azure-pipelines-blueprint-qa-integration.md]: to automate the unite and regression
s/b
to automate the unit and regression

File: content/azure-pipelines-blueprint-qa-integration.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+## Continuous delivery (CD) QA extension points?
+
+![CD Integration](../images/azure-pipelines-blueprint-qa-integration-3.png)
+
+The second extension, which runs in parallel to the security automation scans, can be found in the[qa-scans-cd.yml](https://github.com/WorkSafeBC-Common-Engineering/AzureDevOps.Automation.Pipeline.Templates.v2/blob/master/templates/qa/qa-scans-cd.yml) template.
+
+A few words of caution:
+
+- This extension point re-used and triggered by all blueprints!
```
Comment (MerlinDarkstar) [inline content/azure-pipelines-blueprint-qa-integration.md]: This extension point re-used and
s/b
This extension point is re-used and

File: content/azure-pipelines-blueprint-qa-integration.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+```
+# -----------------------------------------------------------------------------------------------------
+# QA AUTOMATION FOR HIGHER (PROD) ENVIRONMENTS STAGE
+# -----------------------------------------------------------------------------------------------------
+- ${{ if or(eq(variables['Build.SourceBranch'], 'refs/heads/release'), startsWith(variables['Build.SourceBranch'], 'refs/heads/release/')) }}:
+```
+
+In other words, you can selectively run quality-assurance in all pipelines or only in pipelines that are based on the ```release/*``` or ```release*``` source code branches, destined for production.
+
+Within both these sections, you will find **blueprint** specific sections, allowing you to your quality assurance automation scans for selected blueprints only.
```
Comment (MerlinDarkstar) [inline content/azure-pipelines-blueprint-qa-integration.md]: allowing you to your quality assurance automation scans for selected blueprints only.
s/b
allowing you to configure your quality assurance automation scans for the selected blueprints only.

File: content/azure-pipelines-blueprint-qa-integration.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+```
+- ${{ if eq( lower(parameters.applicationBlueprint), 'azure-function' ) }}:
+    - script: echo deal with qa scan relevant to azure-function application type
+```
+
+Once you have decided WHEN and for WHICH blueprints you want to run your quality assurance automation scans, insert the into the [qa-scans-cd.yml](https://github.com/WorkSafeBC-Common-Engineering/AzureDevOps.Automation.Pipeline.Templates.v2/blob/master/templates/qa/qa-scans-cd.yml) template. You **must** honour and deal with the ```modeElite```.
```
Comment (MerlinDarkstar) [inline content/azure-pipelines-blueprint-qa-integration.md]: insert the into the
s/b
insert the following into the

**Review by MerlinDarkstar — COMMENTED** [review]
few minor typo's and missing text

**Review by MerlinDarkstar — APPROVED** [review]

**Review by danielrbroderick — APPROVED** [review]

**Review by littleswancode — APPROVED** [review]

**Pull request merged**